pkg/limits/stream_metadata.go

```diff
@@ -0,0 +1,232 @@
+package limits
+
+import (
+	"hash/fnv"
+	"sync"
+)
+
+// AllFunc is a function that is called for each stream in the metadata.
+// It is used to count per tenant active streams.
+// Note: The collect function should not modify the stream metadata.
```

Comment:
Is modification possible as we pass a copy of `Stream`? I suppose you could change the values in `RateBuckets`?